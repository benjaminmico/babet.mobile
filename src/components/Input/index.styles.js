import styled from 'styled-components'

export const InputContainer = styled.TouchableOpacity`
  border-bottom-width: ${props => (props.theme === 'dark' ? 0 : 1)}px;
  border-color: ${props => (props.theme === 'dark' ? 'transparent' : 'rgba(220, 218, 218, 0.5)')};

  ${'' /* border-color: rgba(220, 218, 218, 0.5); */}
  width: 100%;
  height: 72px;
  padding-top: 14px;
  padding-left: 21.5px;
  padding-bottom: 6px;
`

export const InputLabel = styled.Text`
  color: ${props => props.color};
  font-family: 'AvenirNext-Medium';
  font-size: 13px;
  letter-spacing: -0.14px;
  line-height: 18px;
  text-align: left;
`

export const InputValue = styled.TextInput`
  color: ${props => props.color};
  font-family: 'AvenirNext-DemiBold';
  font-size: 15px;
  letter-spacing: -0.16px;
  line-height: 20px;
  padding: 0px;
  border-width: 0px;
  text-align: left;
`
