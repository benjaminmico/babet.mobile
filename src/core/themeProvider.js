import AsyncStorage from '@react-native-community/async-storage'
import React, {useContext, useEffect, useState} from 'react'
import {useDarkMode} from 'react-native-dark-mode'
import THEMES from './themes.json'

const STORAGE_KEY = 'THEME_ID'
const ThemeContext = React.createContext()

export const ThemeContextProvider = ({children}) => {
  const [themeID, setThemeID] = useState()

  const isDarkMode = useDarkMode ? useDarkMode() : false

  useEffect(() => {
    ;(async () => {
      // eslint-disable-next-line no-unused-expressions
      isDarkMode ? setThemeID(THEMES[0].key) : setThemeID(THEMES[1].key)
      // const storedThemeID = await AsyncStorage.getItem(STORAGE_KEY)
      // if (storedThemeID) setThemeID(storedThemeID)
      // // eslint-disable-next-line no-unused-expressions
      // else isDarkMode ? setThemeID(THEMES[0].key) : setThemeID(THEMES[1].key)
    })()
  }, [])

  return <ThemeContext.Provider value={{themeID, setThemeID}}>{themeID ? children : null}</ThemeContext.Provider>
}

export function withTheme(Component) {
  return props => {
    const {themeID, setThemeID} = useContext(ThemeContext)

    const getTheme = themeID => THEMES.find(theme => theme.key === themeID)
    const setTheme = themeID => {
      AsyncStorage.setItem(STORAGE_KEY, themeID)
      setThemeID(themeID)
    }

    return <Component {...props} themes={THEMES} theme={getTheme(themeID)} setTheme={setTheme} />
  }
}

export function getTheme() {
  const {themeID} = useContext(ThemeContext)

  const getThemeContent = themeID => THEMES.find(theme => theme.key === themeID)

  return getThemeContent(themeID)
}
