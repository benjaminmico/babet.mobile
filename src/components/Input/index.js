// @flow

import {withTheme} from '@core/themeProvider'
import React from 'react'
import {InputContainer, InputLabel, InputValue} from './index.styles'

type Props = {
  // input label
  inputLabel: String,
  // theme
  theme: Object,
}

const Input = ({inputLabel, theme}: Props) => {
  // get theme props
  const {
    colors: {
      texts: {
        input: {label: inputLabelTextColor, value: inputValueTextColor},
      },
    },
  } = theme

  // render input
  return (
    <InputContainer>
      <InputLabel color={inputLabelTextColor}>{inputLabel}</InputLabel>
      <InputValue color={inputValueTextColor} />
    </InputContainer>
  )
}

export default withTheme(Input)
