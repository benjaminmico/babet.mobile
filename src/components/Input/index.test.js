/**
 * @jest-environment jsdom
 */
import React from 'react'
import {StyleSheet} from 'react-native'
import {ThemeContextProvider} from '@core/themeProvider'
import {mount} from 'enzyme'
import Input from '.'

describe.only('Input', () => {
  describe('Integrity tests', () => {
    it('renders Input correctly', () => {
      const inputLabel = 'Pseudo'
      const placeholder = 'Entrez votre pseudo'

      const wrapper = mount(
        <ThemeContextProvider>
          <Input inputLabel={inputLabel} placeholder={placeholder} />
        </ThemeContextProvider>,
      )

      const InputLabel = wrapper.find('Text').first()
      const InputItem = wrapper.find('TextInput').first()

      expect(InputLabel.exists()).toBe(true)
      expect(InputLabel.prop('children')).toBe(inputLabel)
      expect(InputItem.exists()).toBe(true)
      expect(InputItem.prop('placeholder')).toBe(placeholder)
    })
  })

  describe('Props tests', () => {
    it('renders  Input', () => {
      const inputLabel = 'Pseudo'

      const wrapper = mount(
        <ThemeContextProvider>
          <Input inputLabel={inputLabel} />
        </ThemeContextProvider>,
      )
      const InputContainer = wrapper.find('View').first()
      const InputLabel = wrapper.find('Text').first()
      const InputValue = wrapper.find('TextInput').first()

      expect(wrapper).toMatchSnapshot()

      const stylesContainer = StyleSheet.flatten(InputContainer.prop('style'))
      expect(stylesContainer.borderBottomWidth).toBe(1)
      expect(stylesContainer.borderColor).toBe('rgba(220, 218, 218, 0.5)')
      expect(stylesContainer.width).toBe('100%')
      expect(stylesContainer.height).toBe(72)
      expect(stylesContainer.paddingTop).toBe(14)
      expect(stylesContainer.paddingLeft).toBe(21.5)

      const stylesLabel = StyleSheet.flatten(InputLabel.prop('style'))
      expect(stylesLabel.color).toBe('#7C7C7C')
      expect(stylesLabel.fontFamily).toBe('AvenirNext-Medium')
      expect(stylesLabel.fontSize).toBe(13)
      expect(stylesLabel.letterSpacing).toBe(-0.14)
      expect(stylesLabel.lineHeight).toBe(18)
      expect(stylesLabel.textAlign).toBe('left')

      const stylesInputValue = StyleSheet.flatten(InputValue.prop('style'))
      expect(stylesInputValue.color).toBe('#000000')
      expect(stylesInputValue.fontFamily).toBe('AvenirNext-DemiBold')
      expect(stylesInputValue.fontSize).toBe(15)
      expect(stylesInputValue.letterSpacing).toBe(-0.16)
      expect(stylesInputValue.lineHeight).toBe(20)
      expect(stylesInputValue.marginTop).toBe(8)
      expect(stylesInputValue.textAlign).toBe('left')
    })
  })
})
