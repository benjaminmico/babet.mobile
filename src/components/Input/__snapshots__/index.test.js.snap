// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Props tests renders  Input 1`] = `
<ThemeContextProvider>
  <Component
    inputLabel="Pseudo"
  >
    <Input
      inputLabel="Pseudo"
      setTheme={[Function]}
      theme={
        Object {
          "backgroundColor": "#ffffff",
          "color": "#101112",
          "colors": Object {
            "components": Object {
              "background": "#FFFFFF",
              "backgroundAction": "#F3F2F2",
              "backgroundContainer": "#FFFFFF",
              "backgroundFilter": "#F3F2F2",
              "backgroundInputButton": "#5100FF",
              "backgroundPrimaryAction": "#5100FF",
              "icon": "#000000",
            },
            "palette": Object {
              "black": "#000000",
              "comparisonDescription": "#A3A2A2",
              "line": "#D4D8D9",
              "lost": "#CA1010",
              "pending": "#7C7C7C",
              "purple": "#5100FF",
              "white": "#FFFFFF",
              "won": "#14B916",
            },
            "shadows": Object {},
            "texts": Object {
              "buttons": Object {
                "actions": "#5100FF",
                "cancellable": "#3E3E3E",
                "filter": "#3E3E3E",
                "isPrimaryAction": "#FFFFFF",
                "social": Object {
                  "apple": "#000000",
                  "facebook": "#1976D2",
                  "gmail": "#F44336",
                },
              },
              "description": "#ADADAD",
              "filters": Object {
                "multiFilter": "#5100FF",
                "selected": "#000000",
                "unselected": "#A3A2A2",
              },
              "input": Object {
                "label": "#7C7C7C",
                "value": "#000000",
              },
              "labelInput": "#7C7C7C",
              "reverseText": "#FFFFFF",
              "stats": Object {
                "kpiValue": "#3E3E3E",
              },
              "text": "#000000",
              "title": "#000000",
            },
          },
          "isShadow": true,
          "key": "white",
          "statusBar": "dark-content",
        }
      }
      themes={
        Array [
          Object {
            "backgroundColor": "#101112",
            "color": "#ffffff",
            "colors": Object {
              "components": Object {
                "background": "#101112",
                "backgroundAction": "#F3F2F2",
                "backgroundContainer": "#2F3134",
                "backgroundFilter": "#F3F2F2",
                "backgroundInputButton": "#5100FF",
                "backgroundPrimaryAction": "#5100FF",
                "icon": "#FFFFFF",
              },
              "palette": Object {
                "black": "#000000",
                "comparisonDescription": "#A3A2A2",
                "line": "#D4D8D9",
                "lost": "#CA1010",
                "pending": "#7C7C7C",
                "purple": "#5100FF",
                "white": "#FFFFFF",
                "won": "#14B916",
              },
              "texts": Object {
                "buttons": Object {
                  "actions": "#5100FF",
                  "cancellable": "#3E3E3E",
                  "filter": "#3E3E3E",
                  "isPrimaryAction": "#FFFFFF",
                  "social": Object {
                    "apple": "#000000",
                    "facebook": "#1976D2",
                    "gmail": "#F44336",
                  },
                },
                "description": "#ADADAD",
                "filters": Object {
                  "multiFilter": "#5100FF",
                  "selected": "#000000",
                  "unselected": "#A3A2A2",
                },
                "input": Object {
                  "label": "#7C7C7C",
                  "value": "#FFFFFF",
                },
                "reverseText": "#FFFFFF",
                "stats": Object {
                  "kpiValue": "#FFFFFF",
                },
                "text": "#FFFFFF",
                "title": "#FFFFFF",
              },
            },
            "isShadow": false,
            "key": "dark",
            "statusBar": "light-content",
          },
          Object {
            "backgroundColor": "#ffffff",
            "color": "#101112",
            "colors": Object {
              "components": Object {
                "background": "#FFFFFF",
                "backgroundAction": "#F3F2F2",
                "backgroundContainer": "#FFFFFF",
                "backgroundFilter": "#F3F2F2",
                "backgroundInputButton": "#5100FF",
                "backgroundPrimaryAction": "#5100FF",
                "icon": "#000000",
              },
              "palette": Object {
                "black": "#000000",
                "comparisonDescription": "#A3A2A2",
                "line": "#D4D8D9",
                "lost": "#CA1010",
                "pending": "#7C7C7C",
                "purple": "#5100FF",
                "white": "#FFFFFF",
                "won": "#14B916",
              },
              "shadows": Object {},
              "texts": Object {
                "buttons": Object {
                  "actions": "#5100FF",
                  "cancellable": "#3E3E3E",
                  "filter": "#3E3E3E",
                  "isPrimaryAction": "#FFFFFF",
                  "social": Object {
                    "apple": "#000000",
                    "facebook": "#1976D2",
                    "gmail": "#F44336",
                  },
                },
                "description": "#ADADAD",
                "filters": Object {
                  "multiFilter": "#5100FF",
                  "selected": "#000000",
                  "unselected": "#A3A2A2",
                },
                "input": Object {
                  "label": "#7C7C7C",
                  "value": "#000000",
                },
                "labelInput": "#7C7C7C",
                "reverseText": "#FFFFFF",
                "stats": Object {
                  "kpiValue": "#3E3E3E",
                },
                "text": "#000000",
                "title": "#000000",
              },
            },
            "isShadow": true,
            "key": "white",
            "statusBar": "dark-content",
          },
        ]
      }
    >
      <Styled(View)>
        <StyledNativeComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "displayName": "Styled(View)",
              "inlineStyle": InlineStyle {
                "rules": Array [
                  "
  border-bottom-width: 1px;
  border-color: rgba(220, 218, 218, 0.5);
  width: 100%;
  height: 72px;
  padding-top: 11px;
  padding-left: 21.5px;
  padding-bottom: 11px;
",
                ],
              },
              "render": [Function],
              "styledComponentId": "StyledNativeComponent",
              "target": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <View
            style={
              Array [
                Object {
                  "borderBottomWidth": 1,
                  "borderColor": "rgba(220, 218, 218, 0.5)",
                  "height": 72,
                  "paddingBottom": 11,
                  "paddingLeft": 21.5,
                  "paddingTop": 11,
                  "width": "100%",
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderBottomWidth": 1,
                    "borderColor": "rgba(220, 218, 218, 0.5)",
                    "height": 72,
                    "paddingBottom": 11,
                    "paddingLeft": 21.5,
                    "paddingTop": 11,
                    "width": "100%",
                  },
                ]
              }
            >
              <Styled(Text)
                color="#7C7C7C"
              >
                <StyledNativeComponent
                  color="#7C7C7C"
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "displayName": "Styled(Text)",
                      "inlineStyle": InlineStyle {
                        "rules": Array [
                          "
  color: ",
                          [Function],
                          ";
  font-family: 'AvenirNext-Medium';
  font-size: 13px;
  letter-spacing: -0.14px;
  line-height: 18px;
  text-align: left;
",
                        ],
                      },
                      "render": [Function],
                      "styledComponentId": "StyledNativeComponent",
                      "target": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <Text
                    color="#7C7C7C"
                    style={
                      Array [
                        Object {
                          "color": "#7C7C7C",
                          "fontFamily": "AvenirNext-Medium",
                          "fontSize": 13,
                          "letterSpacing": -0.14,
                          "lineHeight": 18,
                          "textAlign": "left",
                        },
                      ]
                    }
                  >
                    <Text
                      color="#7C7C7C"
                      style={
                        Array [
                          Object {
                            "color": "#7C7C7C",
                            "fontFamily": "AvenirNext-Medium",
                            "fontSize": 13,
                            "letterSpacing": -0.14,
                            "lineHeight": 18,
                            "textAlign": "left",
                          },
                        ]
                      }
                    >
                      Pseudo
                    </Text>
                  </Text>
                </StyledNativeComponent>
              </Styled(Text)>
              <Styled(Component)
                color="#000000"
              >
                <StyledNativeComponent
                  color="#000000"
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "State": Object {
                        "blurTextInput": [Function],
                        "currentlyFocusedField": [Function],
                        "focusTextInput": [Function],
                      },
                      "attrs": Array [],
                      "displayName": "Styled(Component)",
                      "inlineStyle": InlineStyle {
                        "rules": Array [
                          "
  color: ",
                          [Function],
                          ";
  font-family: 'AvenirNext-DemiBold';
  font-size: 15px;
  letter-spacing: -0.16px;
  line-height: 20px;
  margin-top: 8px;
  text-align: left;
",
                        ],
                      },
                      "render": [Function],
                      "styledComponentId": "StyledNativeComponent",
                      "target": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <Component
                    allowFontScaling={true}
                    color="#000000"
                    rejectResponderTermination={true}
                    style={
                      Array [
                        Object {
                          "color": "#000000",
                          "fontFamily": "AvenirNext-DemiBold",
                          "fontSize": 15,
                          "letterSpacing": -0.16,
                          "lineHeight": 20,
                          "marginTop": 8,
                          "textAlign": "left",
                        },
                      ]
                    }
                    underlineColorAndroid="transparent"
                  >
                    <TextInput
                      allowFontScaling={true}
                      color="#000000"
                      rejectResponderTermination={true}
                      style={
                        Array [
                          Object {
                            "color": "#000000",
                            "fontFamily": "AvenirNext-DemiBold",
                            "fontSize": 15,
                            "letterSpacing": -0.16,
                            "lineHeight": 20,
                            "marginTop": 8,
                            "textAlign": "left",
                          },
                        ]
                      }
                      underlineColorAndroid="transparent"
                    />
                  </Component>
                </StyledNativeComponent>
              </Styled(Component)>
            </View>
          </View>
        </StyledNativeComponent>
      </Styled(View)>
    </Input>
  </Component>
</ThemeContextProvider>
`;
